frontend/butcher-front/Dockerfile

```diff
@@ -0,0 +1,9 @@
+FROM node:12.16.1-alpine AS builder
+WORKDIR /usr/src/app
+COPY package.json package-lock.json ./
+RUN npm install
+COPY . .
+RUN npm run build --prod
+
+FROM nginx:1.15.8-alpine
+COPY --from=builder /usr/src/app/dist/butcher-front/ /usr/share/nginx/html
```

Comment:
take a look at the `backend/Dockerfile`. same rules appy here.

Reply:
Don't know how to change the user in this image without being a big pain in the ass. Feel free to change it if needed. But I think for some static web resources it's not extremely critical.

Reply:
I've got this somewhere. `alipne` is not that hard. will post it tomorrow.

Reply:
Alpine isn't the problem. It's just this specific nginx-alpine image. But yeah probably you just have to edit some configs like /etc/nginx/nginx.conf.

Reply:
i don't get it. why? 👼

Reply:
Changing it to something like this:
```
FROM nginx:1.15.8-alpine

RUN addgroup -S haubzit && adduser -S haubzit -G haubzit

EXPOSE 80

USER haubzit

COPY --from=builder /usr/src/app/dist/butcher-front/ /usr/share/nginx/html
```
Results in:
```
hlfabric@hlfabric-VirtualBox:~/Documents/git/codebase$ kubectl logs haubzit-frontend-5f7976864b-pkx7c
2020/10/12 20:01:26 [warn] 1#1: the "user" directive makes sense only if the master process runs with super-user privileges, ignored in /etc/nginx/nginx.conf:2
nginx: [warn] the "user" directive makes sense only if the master process runs with super-user privileges, ignored in /etc/nginx/nginx.conf:2
2020/10/12 20:01:26 [emerg] 1#1: mkdir() "/var/cache/nginx/client_temp" failed (13: Permission denied)
nginx: [emerg] mkdir() "/var/cache/nginx/client_temp" failed (13: Permission denied)
```

But according to some posts like this https://stackoverflow.com/questions/42329261/running-nginx-as-non-root-user you can change the /etc/nginx/nginx.conf to the desired user and create+change the ownership of some directories (like /var/cache/nginx/client_temp) that would otherwise be created by root when the container starts. I can try to fix it tomorrow.

Reply:
I just looked the one up I had in mind but recognised it was a plain `node` image. let's skip this for now, as you say, it's probably too much of a pain. althought I have some security conserns.. are you sure, something like "directory traveling" isn't possible? is there something like `.htaccess` for apache?